lib/ansible/plugins/connection/ssh.py

```diff
@@ -677,7 +677,7 @@ def _bare_run(self, cmd, in_data, sudoable=True, checkrc=True):
         # only when using ssh. Otherwise we can send initial data straightaway.
 
         state = states.index('ready_to_send')
-        if b'ssh' in cmd and sudoable:
+        if to_bytes(self.get_option('ssh_executable')) in cmd and sudoable:
```

Comment:
This looks like a good change, however, for consistency with the rest of the code, I think we should use `self._play_context.ssh_executable` until we decide to switch the connection plugin in full over to using `get_option`.

Reply:
in the process of doing so, play_context should not really be the main source of truth anymore